src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -350,7 +357,8 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 	}
 
 	/* check that table is empty if that is required */
-	if (requireEmpty && !LocalTableEmpty(relationId))
+	*emptyTable = LocalTableEmpty(relationId);
```

Comment:
Since we need this information later. Can we establish whether the table is empty prior to calling `ConvertToDistributedTable` and just pass the result as we do with `CreateShardsWithRoundRobinPolicy`?